src/providers/ldap/sdap_sudo_shared.c

```diff
@@ -129,25 +129,17 @@ sdap_sudo_ptask_setup_generic(struct be_ctx *be_ctx,
 static char *
 sdap_sudo_new_usn(TALLOC_CTX *mem_ctx,
                   unsigned long usn,
-                  const char *leftover,
-                  bool supports_usn)
+                  const char *leftover)
 {
     const char *str = leftover == NULL ? "" : leftover;
     char *newusn;
 
-    /* This is a fresh start and server uses modifyTimestamp. We need to
-     * provide proper datetime value. */
-    if (!supports_usn && usn == 0) {
-        newusn = talloc_strdup(mem_ctx, "00000101000000Z");
-        if (newusn == NULL) {
-            DEBUG(SSSDBG_MINOR_FAILURE, "Unable to change USN value (OOM)!\n");
-            return NULL;
-        }
-
-        return newusn;
+    /* Current largest USN is unknown so we keep "0" to indicate it. */
+    if (usn == 0) {
+        return talloc_strdup(mem_ctx, "0");
     }
 
-    /* We increment USN number so that we can later use simplify filter
+    /* We increment USN number so that we can later use simplified filter
      * (just usn >= last+1 instead of usn >= last && usn != last).
      */
     usn++;
```

Comment:
Previously `usn` was incremented in case (it was 0 and `supports_usn == true`).
Now it will be kept "0" in this case.
Is this intentional?

Reply:
Yes. It does not make any sense to search for usn >= 1 since everything would satisfy this condition. Now we keep 0, which means "do not add usn/modifytimestamp to the filter". We keep 0 until we get any meaningful value.

Reply:
This was the root cause of the problem - with usn value you have server-wide lastusn so we did not really use usn>=1 in filter. But there is no global last modifyTimestamp so we kept searching with 1 until we got at least one rule which caused performance issues because anything passed this condition.